.github/workflows/cypress.yml

```
@@ -118,6 +130,53 @@ jobs:
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: 'us-gov-west-1'
 
+      - name: Retrieve Failure State
+        uses: nick-invision/persist-action-data@v1
+        with:
+          data: ${{ env.TEST_FAILURE }}
+          retrieve_variables: TEST_FAILURE
+
+      - name: Get Slack app token
+        if: ${{ env.TEST_FAILURE }}
+        uses: marvinpinto/action-inject-ssm-secrets@v1.1.1
+        with:
+          ssm_parameter: /devops/github_actions_slack_socket_token
+          env_variable_name: SLACK_APP_TOKEN
+
+      - name: Get Slack bot token
+        if: ${{ env.TEST_FAILURE }}
+        uses: marvinpinto/action-inject-ssm-secrets@v1.1.1
+        with:
+          ssm_parameter: /devops/github_actions_slack_bot_user_token
+          env_variable_name: SLACK_BOT_TOKEN
+
+      - name: Get va-vsp-bot token
+        uses: marvinpinto/action-inject-ssm-secrets@v1.1.1
+        with:
+          ssm_parameter: /devops/VA_VSP_BOT_GITHUB_TOKEN
+          env_variable_name: VA_VSP_BOT_GITHUB_TOKEN
+
+      - name: Checkout VSP actions
+        if: ${{ env.TEST_FAILURE }}
+        uses: actions/checkout@v2
+        with:
+          repository: department-of-veterans-affairs/vsp-github-actions
+          ref: refs/heads/main
+          token: ${{ env.VA_VSP_BOT_GITHUB_TOKEN }}
+          persist-credentials: false
+          path: ./.github/actions/vsp-github-actions
+
+      - name: Notify Slack
+        if: ${{ env.TEST_FAILURE }}
```

I don't think you need to persist a variable between jobs. Could probably do:

```suggestion
        if: ${{ needs.build.result == 'failure' }}
```

---

Thanks for this.  This was a great suggestion as it now only gives the notification on a test failure such as not passing an assertion, instead of ANY failure (prior, it was sending the notification just on any build error, which is not what we were after).  This worked great! (and I learned something new)